### PR TITLE
port: strip_auth middleware to tip

### PR DESCRIPTION
**Pull request by asoorm**
refs #1273 

TODO:

- [x] Write Tests
- [x] Establish how the mw should react on error `stripFromParams` (Ignore errors)
- [x] confirm where `mwAppendEnabled` should be placed (Same as Martin's example)

### CODE REVIEW

File: mw_strip_auth.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+	return nil, 200
+}
+
+func (sa *StripAuth) stripFromParams(r *http.Request) error {
+
+	config := sa.Spec.Auth
+
+	reqUrlPtr, err := url.Parse(r.URL.String())
+	if err != nil {
+		return err
```
Comment (buger) [inline mw_strip_auth.go]: I think it should be totally safe ignore both of this errors in this function. In the first case, you are parsing current URL, which was already validated. In the second case, almost the same URL object, just changed query string a bit.

File: mw_strip_auth_test.go [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+}
+
+func TestStripAuth_Name(t *testing.T) {
+
+	sa := StripAuth{}
+
+	if sa.Name() != "StripAuth" {
+		t.Error("Incorrect Name")
+	}
+}
```
Comment (mvdan) [inline mw_strip_auth_test.go]: This is a bit unnecessary.

File: mw_strip_auth_test.go [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+	if sa.EnabledForSpec() != sa.Spec.StripAuthData {
+		t.Error("EnablesForSpec be true when sa.Spec.StripAuthData is true")
+	}
+
+	sa.Spec.StripAuthData = false
+
+	if sa.EnabledForSpec() != sa.Spec.StripAuthData {
+		t.Error("EnablesForSpec should false when sa.Spec.StripAuthData is false")
+	}
+}
```
Comment (mvdan) [inline mw_strip_auth_test.go]: This one too - you're writing a dozen lines of test for a single line of trivial code.

If you want to be sure that the middleware works, don't unit test every single part of it. Unit test the complex bits like you do below, and then do a full end-to-end test. Should be easy with httptest and loading a custom spec.

**Comment by buger** [comment]
Looks good! Lets test together with UI part.

**Comment by mvdan** [comment]
@asoorm please rebase - we need to merge this in for 2.4

**Comment by asoorm** [comment]
@mvdan Done

**Pull request merged**